backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramCreationService.java

```diff
@@ -116,7 +123,15 @@ private Diagram doRender(String label, Object targetObject, IEditingContext edit
         variableManager.put(VariableManager.SELF, targetObject);
         variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
 
-        DiagramComponentProps props = new DiagramComponentProps(variableManager, diagramDescription, viewCreationRequests, optionalPreviousDiagram);
+        Map<UUID, Position> movedElementIdToNewPositionMap = optionalDiagramContext.map(IDiagramContext::getMovedElementIDToNewPositionMap).orElseGet(() -> Map.of());
+        Set<UUID> allMovedElementIds = Set.of();
+        if (!movedElementIdToNewPositionMap.isEmpty() && optionalDiagramContext.isPresent()) {
+            allMovedElementIds = this.getRelatedElementIDs(movedElementIdToNewPositionMap.keySet(), optionalDiagramContext.get().getDiagram());
+        }
+        Optional<Diagram> optionalPreviousDiagram = optionalDiagramContext.map(IDiagramContext::getDiagram);
+        Optional<Position> optionalStartingPosition = Optional.ofNullable(optionalDiagramContext.map(IDiagramContext::getStartingPosition).orElse(null));
+        DiagramComponentProps props = new DiagramComponentProps(variableManager, diagramDescription, viewCreationRequests, optionalPreviousDiagram, movedElementIdToNewPositionMap, allMovedElementIds,
+                optionalStartingPosition);
```

Comment:
The whole thing (including the methods below such as getRelatedElementIDs) is convoluted. After more than 15min on them, I still don't know for sure what they are supposed to do. I am still not convinced that you need those two data structures especially since one is derived from the other. I am still not sure why the first one is a map since you seem to put only one element inside in `UpdateNodePositionEventHandler.handleUpdateNodePosition` before removing it when the render is done.

So, as stated multiple time before, write an ADR to explain what you are doing.

Reply:
Fixed

Reply:
ADR and fix will come in the next commit